src/viadot/orchestration/prefect/utils.py

```diff
@@ -28,6 +28,394 @@
+    ) -> List[str]:
+        """
+        Generate a list of years either for the last X years or from a start year.
+
+        Args:
+            last_years (int): The number of years from the current year to include.
+            from_year (str): The starting year.
+            num_years (int): The number of years to generate from the starting year.
+
+        Returns:
+            list: A list of years in ascending order.
+        """
+        current_year = pendulum.now().year
+        if last_years:
+            result = [str(current_year - i) for i in range(last_years)][
+                ::-1
+            ]  # Reversed to ascending order
+            return result
+        elif from_year and num_years:
+            start_year = int(from_year)
+            result = [str(start_year - i) for i in range(num_years)][
+                ::-1
+            ]  # Ascending order
+            return result
+        return []
+
+    def generate_months(self, last_months: int) -> List[str]:
+        """
+        Generate a list of first days of the last X months.
+
+        Args:
+            last_months (int): The number of months to include from the past.
+
+        Returns:
+            list: A list of dates representing the first day of the last X months in ascending order.
+        """
+        current_date = pendulum.now()
+        result = [
+            current_date.subtract(months=i).start_of("month").format("YMM")
+            for i in range(last_months)
+        ][
+            ::-1
+        ]  # Reversed to ascending order
+        return result
+
+    def generate_dates(self, last_days: int) -> List[str]:
+        """
+        Generate a list of dates for the last X days.
+
+        Args:
+            last_days (int): The number of days to include from the past.
+
+        Returns:
+            list: A list of dates in ascending order.
+        """
+        current_date = pendulum.now(self.dynamic_date_timezone)
+        result = [
+            current_date.subtract(days=i).format("YMMDD") for i in range(last_days)
+        ][
+            ::-1
+        ]  # Reversed to ascending order
+        return result
+
+    def process_first_days(
+        self, month_name: str, year: int, num_days: int
+    ) -> List[str]:
+        """
+        Generate a list of the first X days of a given month and year.
+
+        Args:
+            month_name (str): The name of the month.
+            year (str): The year.
+            num_days (int): The number of days to include.
+
+        Returns:
+            list: A list of dates for the first X days in ascending order.
+        """
+        start_date = pendulum.date(
+            int(year), pendulum.parse(month_name, strict=False).month, 1
+        )
+        result = [
+            start_date.add(days=i).format("YMMDD") for i in range(num_days)
+        ]  # Ascending order
+        return result
+
+    def process_last_days(self, month_name: str, year: int, num_days: int) -> List[str]:
```

Comment:
mark this function internal by adding `_`.
```suggestion
    def _process_last_days(self, month_name: str, year: int, num_days: int) -> List[str]:
```